mithril-aggregator/src/database/provider/signed_entity/get_signed_entity.rs

```diff
@@ -0,0 +1,171 @@
+        Self { client }
+    }
+
+    fn condition_by_signed_entity_id(&self, signed_entity_id: &str) -> StdResult<WhereCondition> {
+        Ok(WhereCondition::new(
+            "signed_entity_id = ?*",
+            vec![Value::String(signed_entity_id.to_owned())],
+        ))
+    }
+
+    fn condition_by_certificate_id(&self, certificate_id: &str) -> StdResult<WhereCondition> {
+        Ok(WhereCondition::new(
+            "certificate_id = ?*",
+            vec![Value::String(certificate_id.to_owned())],
+        ))
+    }
+
+    fn condition_by_certificates_ids(&self, certificates_ids: &[&str]) -> WhereCondition {
+        let ids_values = certificates_ids
+            .iter()
+            .map(|id| Value::String(id.to_string()))
+            .collect();
+
+        WhereCondition::where_in("certificate_id", ids_values)
+    }
+
+    fn condition_by_signed_entity_type(
+        &self,
+        signed_entity_type: &SignedEntityTypeDiscriminants,
+    ) -> StdResult<WhereCondition> {
+        let signed_entity_type_id: i64 = signed_entity_type.index() as i64;
+
+        Ok(WhereCondition::new(
+            "signed_entity_type_id = ?*",
+            vec![Value::Integer(signed_entity_type_id)],
+        ))
+    }
+
+    /// Get SignedEntityRecords for a given signed_entity id.
+    pub fn get_by_signed_entity_id(
+        &self,
+        signed_entity_id: &str,
+    ) -> StdResult<EntityCursor<SignedEntityRecord>> {
+        let filters = self.condition_by_signed_entity_id(signed_entity_id)?;
+        let signed_entity_record = self.find(filters)?;
+
+        Ok(signed_entity_record)
+    }
+
+    /// Get [record][SignedEntityRecord] for a given `certificate_id`.
+    pub fn get_by_certificate_id(
+        &self,
+        certificate_id: &str,
+    ) -> StdResult<EntityCursor<SignedEntityRecord>> {
+        let filters = self.condition_by_certificate_id(certificate_id)?;
+        let signed_entity_record = self.find(filters)?;
+
+        Ok(signed_entity_record)
+    }
+
+    /// Get [records][SignedEntityRecord] for a list of given `certificates_ids`.
+    pub fn get_by_certificates_ids(
+        &self,
+        certificates_ids: &[&str],
+    ) -> StdResult<EntityCursor<SignedEntityRecord>> {
+        let filters = self.condition_by_certificates_ids(certificates_ids);
+        let signed_entity_record = self.find(filters)?;
+
+        Ok(signed_entity_record)
+    }
+
+    /// Get SignedEntityRecords for a given signed entity type.
+    pub fn get_by_signed_entity_type(
+        &self,
+        signed_entity_type: &SignedEntityTypeDiscriminants,
+    ) -> StdResult<EntityCursor<SignedEntityRecord>> {
+        let filters = self.condition_by_signed_entity_type(signed_entity_type)?;
+        let signed_entity_record = self.find(filters)?;
+
+        Ok(signed_entity_record)
+    }
+
+    #[cfg(test)]
+    /// Get all SignedEntityRecords.
+    pub fn get_all(&self) -> StdResult<EntityCursor<SignedEntityRecord>> {
```

Comment:
Agree, but this should go with a refactor of the provider system to separate CRUD and GET operations. Else it could add confusion to an already confusing code (since the simplest way of doing this would be to add a blanket method to the `Provider` trait).